ui/src/app/pages/login/account-creation/account-creation.spec.tsx

```diff
@@ -103,128 +97,116 @@ test.each([
-  );
-  expect(findCountryInputField(wrapper).exists()).toBeTruthy();
-  findCountryInputField(wrapper).simulate('change', 'Canada');
-  expect(findCountryInputField(wrapper).prop('value')).toEqual('Canada');
+it('should display a text input field for non-US countries', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('[data-test-id="country-input"]')).toBeNull();
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.unitedStates);
+  await user.keyboard('{enter}');
+  expect(container.querySelector('[data-test-id="country-input"]')).toBeNull();
+  await user.clear(findCountryDropdownField());
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.other);
+  await user.keyboard('{enter}');
+  expect(findCountryInputField()).not.toBeNull();
+  await user.type(findCountryInputField(), 'Canada');
+  expect(findCountryInputField().value).toEqual('Canada');
 });
 
-it('should capitalize a state code when selecting USA', function () {
-  const wrapper = shallowComponent();
-  findStateField(wrapper).simulate('change', 'ny');
-  expect(findStateField(wrapper).prop('value')).toEqual('ny');
-  findCountryDropdownField(wrapper).simulate(
-    'change',
-    countryDropdownOption.unitedStates
-  );
-  expect(findStateField(wrapper).prop('value')).toEqual('NY');
+it('should capitalize a state code when selecting USA', async () => {
+  const user = userEvent.setup();
+  component();
+  await user.type(findStateField(), 'ny');
+  expect(findStateField().value).toEqual('ny');
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.unitedStates);
+  await user.keyboard('{enter}');
+  expect(findStateField().value).toEqual('NY');
 });
 
-it('should change a state name to a state code after selecting USA', function () {
-  const wrapper = shallowComponent();
-  findStateField(wrapper).simulate('change', 'new york');
-  expect(findStateField(wrapper).prop('value')).toEqual('new york');
-  findCountryDropdownField(wrapper).simulate(
-    'change',
-    countryDropdownOption.unitedStates
-  );
-  expect(findStateField(wrapper).prop('value')).toEqual('NY');
+it('should change a state name to a state code after selecting USA', async () => {
+  const user = userEvent.setup();
+  component();
+  await user.type(findStateField(), 'new york');
+  expect(findStateField().value).toEqual('new york');
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.unitedStates);
+  await user.keyboard('{enter}');
+  expect(findStateField().value).toEqual('NY');
 });
 
-it('should mark US states as invalid if not a 2-letter code', function () {
-  const wrapper = shallowComponent();
-  expect(wrapper.exists('#stateError')).toBeFalsy();
-  expect(
-    (wrapper.instance() as AccountCreation).validate()['address.state']
-  ).not.toContain(stateCodeErrorMessage);
```

Comment:
This was replaced with `screen.queryByText(stateCodeErrorMessage)`, which I think is an improvement.